### PR TITLE
feat: Transient Data Policy Validator

### LINKED ISSUE #78: Transient Data Policy Validator
Implement a policy validator for Transient Data collections.

### PR DESCRIPTION
**Pull request by bstasyszyn**
Implemented a policy validator for Transient Data collections.

closes #78

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>

### CODE REVIEW

**Comment by bstasyszyn** [comment]
verify-unitest

**Comment by bstasyszyn** [comment]
verify-unitest

**Comment by bstasyszyn** [comment]
verify-unitest

File: mod/peer/collections/policy/validator.go [hunk 1]
```diff
@@ -7,24 +7,74 @@ SPDX-License-Identifier: Apache-2.0
...
+// NewValidator returns a new collection policy validator
 func NewValidator() *Validator {
 	return &Validator{}
 }
 
 // Validate validates various collection config types
 func (v *Validator) Validate(collConfig *common.CollectionConfig) error {
-	return nil
+	config := collConfig.GetStaticCollectionConfig()
+	if config == nil {
```
Comment (troyronda) [inline mod/peer/collections/policy/validator.go]: Unfortunate that a Boolean isn’t returned.

File: mod/peer/collections/policy/validator.go [hunk 2]
```diff
@@ -7,24 +7,74 @@ SPDX-License-Identifier: Apache-2.0
...
+		return errors.New("unknown collection configuration type")
+	}
+
+	switch config.Type {
+	case common.CollectionType_COL_TRANSIENT:
+		return tdatapolicy.ValidateConfig(config)
+	default:
+		// Nothing to do
+		return nil
```
Comment (troyronda) [inline mod/peer/collections/policy/validator.go]: It would be better if NewValidator didn’t return nil without a validity indicator.

Comment (bstasyszyn) [reply mod/peer/collections/policy/validator.go]: Not sure I know what you mean. Returning nil means that it is valid. This validator only handles extensions to the collection config. All of the basic checks have already been performed.

Comment (troyronda) [reply mod/peer/collections/policy/validator.go]: Never mind :). All good.

**Pull request merged**